cpp/src/arrow/engine/substrait/serde_test.cc

```diff
@@ -724,5 +728,99 @@ TEST(Substrait, ExtensionSetFromPlan) {
   EXPECT_EQ(decoded_add_func.name, "add");
 }
 
+TEST(Substrait, GetRecordBatchReader) {
+  const auto parquet_root = std::getenv("PARQUET_TEST_DATA");
```

Comment:
There is `GetEnvVar` in `io_util.cc` that returns `Result<std::string>` which you can `ASSERT_OK_AND_ASSIGN`.  That might lead to slightly clearer errors in case the test data environment variable is not set.  As it stands `getenv` will return `nullptr` and then constructing a string with it will trigger `std::exception`.

Reply:
Good catch! Will update.